db/builder.cc

```diff
@@ -30,14 +30,12 @@ Status BuildTable(const std::string& dbname, Env* env, const Options& options,
 
     TableBuilder* builder = new TableBuilder(options, file);
     meta->smallest.DecodeFrom(iter->key());
-    Slice key;
     for (; iter->Valid(); iter->Next()) {
-      key = iter->key();
-      builder->Add(key, iter->value());
-    }
-    if (!key.empty()) {
```

Comment:
Unrelated to your change: this check assumes that `iter->Valid()` returned `false` due to having not more data in the iterator, but what if `iters->status()` is not OK? That's an error.